### PR TITLE
Fix some cppcheck warnings

### PR DESCRIPTION
**Pull request by heckflosse**
More fixes will follow...

### CODE REVIEW

File: rtengine/coord.h [hunk 1]
```diff
@@ -114,22 +114,22 @@ inline void Coord::set (const int x, const int y)
 
 inline Coord& Coord::operator+= (const Coord& other)
 {
-    x += other.x;
-    y += other.y;
+    this->x += other.x;
+    this->y += other.y;
```
Comment (Floessie) [inline rtengine/coord.h:118]: @heckflosse What was the warning? I see no obvious reason for this change...

File: rtengine/imagefloat.h [hunk 2]
```diff
@@ -218,14 +218,14 @@ class Imagefloat : public IImagefloat, public ImageIO
...
-    void                 ExecCMSTransform(cmsHTRANSFORM hTransform);
-    void                 ExecCMSTransform(cmsHTRANSFORM hTransform, const LabImage &labImage, int cx, int cy);
+    void normalizeFloat(float srcMinVal, float srcMaxVal) override;
+    void normalizeFloatTo1();
+    void normalizeFloatTo65535();
+    void calcCroppedHistogram(const ProcParams &params, float scale, LUTu & hist);
+    void ExecCMSTransform2(cmsHTRANSFORM hTransform);
+
+    void ExecCMSTransform(cmsHTRANSFORM hTransform);
+    void ExecCMSTransform(cmsHTRANSFORM hTransform, const LabImage &labImage, int cx, int cy);
```
Comment (Floessie) [inline rtengine/imagefloat.h:228]: :+1: If you like, I can fire up clang-tidy to add `override`s everywhere.

Comment (heckflosse) [reply rtengine/imagefloat.h:228]: Please do. :+1:

Comment (Floessie) [reply rtengine/imagefloat.h:228]: Processing...

File: rtengine/coord.h (hunk 1 above)
Comment (heckflosse) [reply rtengine/coord.h:118]: It was a false positive
`Assignment of function parameter has no effect outside the function.`
But using `this->` silenced it.

Comment (Floessie) [reply rtengine/coord.h:118]: We shouldn't code against bugs in cppcheck...

File: rtengine/imagefloat.h (hunk 2 above)
Comment (Floessie) [reply rtengine/imagefloat.h:228]: [Here](https://filebin.net/j87h1dkc7lb65ej2) it is. Take it with a grain of salt, though: `clang-tidy` has some built-in default checks you can't disable. So there's always a bit more done than expected. See the grouping of `NAN`s in `dcraw.cc` in the patch, where one grouping was missed.

HTH,
Flössie

Comment (heckflosse) [reply rtengine/imagefloat.h:228]: @Floessie Great. Shall I push?

File: rtengine/coord.h (hunk 1 above)
Comment (heckflosse) [reply rtengine/coord.h:118]: agreed

File: rtengine/imagefloat.h (hunk 2 above)
Comment (Floessie) [reply rtengine/imagefloat.h:228]: @heckflosse Ingo, sure. Maybe also add the braces around the first `NAN` block in `dcraw.cc` to make it symmetric.

Comment (heckflosse) [reply rtengine/imagefloat.h:228]: Ok, I'm just compiling it before I will push it...

**Pull request closed without merging**